src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -434,11 +460,28 @@ public string DownloadFile(string bucketName, string storageKeyOfFile)
 			}
 		}
 
-		private bool AvoidThisFile(string objectKey)
+		internal static bool AvoidThisFile(string objectKey)
 		{
 			// Note that Amazon S3 regards "/" as the directory delimiter for directory oriented
 			// displays of object keys.
-			return objectKey.ToLowerInvariant().EndsWith("/thumbs.db") || objectKey.ToLowerInvariant().EndsWith(".pdf");
+			string[] endsToAvoid = { "/thumbs.db", ".pdf", ".map"};
+			if (endsToAvoid.Any(end => objectKey.ToLowerInvariant().EndsWith(end)))
+				return true;
+
+			// We only want to download audio for "music", not narration.
+			// The way we determine the difference is that narration audio files are guids.
+			// This isn't 100% accurate because, in theory, someone could choose a music file which has a guid file name.
+			// But we are living with that possibility for now.
+			if (MusicApi.MusicFileExtensions.Any(end => objectKey.ToLowerInvariant().EndsWith(end)))
+			{
+				var match = RegexFileNameRegex.Match(objectKey);
+				if (match.Success)
+				{
+					Guid dummy;
+					return Guid.TryParse(match.Groups[2].Value, out dummy);
```

Comment:
I bear more bad news :(

If you look at the ID generation code in https://github.com/sujeffreyl/BloomDesktop/blob/master/src/BloomBrowserUI/bookEdit/toolbox/talkingBook/audioRecording.ts, line 1011, you can see that the majority of GUIDs will actually end up prefixed by an 'i'.  Guid.TryParse will return false in this case, which means we will upload the narration even though it says not to.  We need to do a 2nd round and strip out the leading 'i'